src/kiota/KiotaHost.cs

```diff
@@ -576,6 +576,22 @@ private static void AddStringRegexValidator(Option<string> option, Regex validat
                 input.ErrorMessage = $"{value} is not a valid {parameterName} for the client, the {parameterName} must conform to {validator}";
         });
     }
+
+    internal static void ValidateBothOrNoneOptions(CommandResult commandResult, Option option1, Option option2)
+    {
+        var option1Result = commandResult.Children.FirstOrDefault(c => c.Symbol == option1) as OptionResult;
+        var option2Result = commandResult.Children.FirstOrDefault(c => c.Symbol == option2) as OptionResult;
```

Comment:
```suggestion
        if (commandResult.Children.FirstOrDefault(c => c.Symbol == option1) is not OptionResult option1Result) return;
        if (commandResult.Children.FirstOrDefault(c => c.Symbol == option2) is not OptionResult option2Result) return;
```

Reply:
(or maybe a combined version of that if you want to validate that both are passed or neither)

Reply:
Update to more general version